core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -267,6 +310,96 @@ object UnitAutomation {
         unit.movement.headTowards(settlerOrGreatPersonToAccompany.currentTile)
         return true
     }
+    
+    private fun tryPatrolBoarders(unit: MapUnit): Boolean {
+        if (unit.type == UnitType.Scout) return false
+
+        val friendlyCitiesTiles = unit.civInfo.cities
+                .filter { unit.civInfo == it.civInfo }
+                .flatMap { it.tiles }
+                .asSequence().map { unit.currentTile.tileMap.get(it) }
+        
+        val weakestInfluenceTile = friendlyCitiesTiles
+                .filter { it.getInfluence() > 0 && (it in unit.civInfo.viewableTiles) }
+                .sortedByDescending { it.getInfluence() }
+                .firstOrNull { unit.movement.canReach(it) }
+        
+        if (weakestInfluenceTile != null) {
+            if (unit.type.isLandUnit() && weakestInfluenceTile.isWater) return false
+            unit.movement.headTowards(weakestInfluenceTile)
+            return true
+        }
+        return false
+    }
+    
+    private fun tryTakeBackCapturedCity(unit: MapUnit): Boolean {
+        var capturedCities = unit.civInfo.gameInfo.civilizations
+                .filter { unit.civInfo.isAtWarWith(it) }
+                .flatMap { it.cities }.asSequence()
+                .filter { unit.civInfo.civName == it.foundingCiv && it.isInResistance() && it.health < it.getMaxHealth() } //Most likely just been captured
+                .asSequence()
+
+        if (unit.type.isRanged()) // ranged units don't harm capturable cities, waste of a turn
+            capturedCities = capturedCities.filterNot { it.health == 1 }
+
+        val closestReachableCapturedCity = capturedCities
+                .asSequence().map { it.getCenterTile() }
+                .filter { unit.movement.canReach(it) }
```

Comment:
Ahhhh noooo don't use canReach before you've sorted by aerialDistanceTo, it's super long
Instead .sortedBy{ it.aerialDistanceTo(unit.getTile()) }.firstOrNull{  unit.movement.canReach(it) }